### DISCUSSION TITLE
Add a display_name to improve foreign key visualization

### DISCUSSION THREAD

**Discussion by JonathanPicques** [supabase/supabase#6932] [Feature Requests]
# Feature request

It could be nice to have a display_name for a linked field

![image](https://user-images.githubusercontent.com/1839757/168811959-63274c6b-4ddf-4a54-822e-523b8e372eb9.png)
Instead of that we could have
![image](https://user-images.githubusercontent.com/1839757/168812644-bd71341f-0948-4c8a-a008-20beca42a58c.png)

## Is your feature request related to a problem? Please describe.

My use case is to hand a supabase account to a client and let them manage their coupons.
As they are not really tech-savyy, the interface is a bit cumbersome to work with when it comes to foreign keys

## Describe the solution you'd like

The ability to set a display name somewhere when you set up a foreign key

**Comment by kiwicopple** [comment]
Hey @JonathanPicques - would a database view for your use-case?

Either way, we might end up doing something like this beyond a database view. I'm going to convert this to a Discussion, because it would be good to gather the community's feedback on this one too.

**Reply by JonathanPicques** [reply]
Hi @kiwicopple thanks for reaching out :)

It's a good solution for making an overview.

The real problem here is adding/updating these values, for instance the field `partner.user_id` is linked to `auth.user.id` and selecting an uid is not user-friendly, I'd rather select the user email in that case

![image](https://user-images.githubusercontent.com/1839757/169803095-145c1f0d-87fe-4814-b266-7503f6f134ce.png)

**Comment by segheysens** [comment]
Love this idea, I agree that it would make updating related records in the Table view easier to use.

**Comment by woutersteven** [comment]
Seconding this, I believe it is done in PHPMyAdmin in the same way, it's very useful.
